test/autorest/generated/lrogroup/pollers.go

```diff
@@ -7,85 +7,48 @@ package lrogroup
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
+	"github.com/Azure/azure-sdk-for-go/sdk/armcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"net/http"
-	"net/url"
 	"time"
 )
 
 // HTTPPoller provides polling facilities until the operation completes
 type HTTPPoller interface {
 	Done() bool
 	Poll(ctx context.Context) (*http.Response, error)
-	FinalResponse() *http.Response
+	FinalResponse(ctx context.Context) (*http.Response, error)
```

Comment:
Why did this change?  Do we really need to perform a final GET for *http.Response operations?

Reply:
We do not :( But we also do not expose the latest response method for the pollers :/

Reply:
I wonder, can we just omit FinalResponse for HTTPPoller types?

Reply:
I suppose that's feasible. I mean we do not perform an additional get for the final http.Response, but it does mean it will be special cased for people who decide to hand write the code without PollUntilDone. We would have to be clear in the documentation that Http Pollers will be different.

Reply:
I see your point, however I personally think this is acceptable.  Having to call FinalResponse() in this case (which does nothing), just for sake of consistency, seems a bit odd to me.

@JeffreyRichter what are your thoughts.

Reply:
Per internal conversation will leave as-is for consistency and versioning.